manager/src/main/java/com/redhat/service/bridge/manager/models/Work.java

```diff
@@ -0,0 +1,120 @@
+package com.redhat.service.bridge.manager.models;
+
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Version;
+
+@NamedQueries({
+        @NamedQuery(name = "Work.findByManagedResourceId", query = "from Work w where w.managedResourceId=:managedResourceId"),
+        @NamedQuery(name = "Work.findByWorkerId", query = "from Work w where w.workerId=:workerId"),
+        @NamedQuery(name = "Work.updateWorkerId", query = "update Work w set w.workerId=:workerId where w.submittedAt >= :age")
```

Comment:
@manstis Should this be checking `modifiedAt` instead of `submittedAt`?

Reply:
That depends if the time-out is based on when the work was submitted or last updated.

I've assumed it to be from when it was first submitted.

Reply:
Review note: await advice from @robpblake before resolving.

Reply:
@manstis I think it should be checking `modifiedAt` e.g if a worker hasn't done anything to a work item for "x" minutes, assume the worker is dead and we should reassign it.

Does that make sense to you or do you feel submitted is the better timestamp to perform timeout on?

Reply:
I can make it as you wish.. I have no **strong** preference. I'll need to check on when/where we update the modified date. Personally, I still think it should be from the submitted timestamp... as a User I'd expect the timeout to be from the moment I asked for it and not, due to implementation details, some seemingly random time in the future. For example, if I open a `Socket` in code I'd expect it to timeout from the moment I request it and not an unpredictable time after that because the "OSI Model - Network Layer 3" took 10 retires to find a route. Either way, if I add JavaDoc saying timeout is per-retry it's clear to everyone... of course, per-retry makes failing things based on SLOs harder...

Reply:
@robpblake WDYT? Still want me to proceed with using `modifiedDate`?

Reply:
@manstis I just want to make sure we're talking about the same thing:

I'm referring here to when a Worker reassigns Work to itself because another worker has "died", so specifically this query:

`
update Work w set w.workerId=:workerId where w.submittedAt >= :age
`

To me that feels more natural to re-assign work to another worker if there is no evidence the work has been "updated" during the last 'x' minutes. If work was submitted 5 minutes ago, but the last update on it was 10 seconds ago, I'd be confident that the worker is still alive and doing its thing.

I am in total agreement with you that from a customer of OpenBridge perspective, the timeout is from the point when I submitted the request to have my Bridge/Processor deployed.

Does that make sense?

Reply:
Ah, yes, fair enough. We were talking about different things. Weird your original comment is exactly on the SQL relating to reassignment whereas my mind was on the timeout check in the `Worker`. I'll change the `update` to use the modified date and check it is actually being modified. I'll also need to update the submitted date for the timeout check in the `Action` to still work correctly (it checks if "`now`-`timeout` is after `submitted`").. which then begs the question. If the submitted date is being amended did we want an immutable `originalSubmittedDate` for SLO purposes... or is that the subject of a different JIRA/discusion?